### PR TITLE
Remove support for Calico CNI and add support for IPv6

### PR DESCRIPTION
**Pull request by betabandido**
Amazon VPC CNI already supports high pod density on EKS (see https://aws.amazon.com/blogs/containers/amazon-vpc-cni-increases-pods-per-node-limits/), so there is no need anymore for using Calico CNI.

While high pod density is supported for IPv4 clusters, some configuration aspects are simpler when using IPv6. Therefore, this MR adds support for creating IPv6-based node groups, as well as enabling the high pod density feature.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -72,4 +72,4 @@ variable "encrypt_ebs" {
 variable "volume_type" {
   type    = string
   default = "gp3"
```
Comment (xorxsan) [inline variables.tf:74]: Do we know if there is a new volume type better than this `gp3`, like a `gp4`?

File: variables.tf [hunk 2]
```diff
@@ -64,6 +59,11 @@ variable "subnet_ids" {
   type = list(string)
 }
 
+variable "enable_high_pod_density" {
```
[the remark this answers is not in the document]
Comment (betabandido) [reply variables.tf:62]: A way to enable the high pod density feature. There's a link in the MR description. We could add a description for the variable.

File: variables.tf (hunk 1 above)
Comment (betabandido) [reply variables.tf:74]: No idea. But, we can do that in another MR.

File: data/userdata.tpl [hunk 3]
```diff
@@ -13,31 +13,9 @@ Content-Type: text/x-shellscript; charset="us-ascii"
...
     --apiserver-endpoint '${cluster_endpoint}' \
     ${bootstrap_extra_args} \
     '${cluster_name}'
 
-if ${disable_source_dest_checks}
-then
-    # Disable source/dest checks
-    # (see https://docs.projectcalico.org/reference/public-cloud/aws#routing-traffic-within-a-single-vpc-subnet)
-
-    export TOKEN=$(
```
[the remark this answers is not in the document]
Comment (xorxsan) [reply data/userdata.tpl:27]: In the launch-template file there is this `data "aws_ami" "ami"` resource which contains the following code:
```
metadata_options {
    http_endpoint = "enabled"
    http_tokens = "required"
    ...
}
```

I thought it was called private endpoints, but it's actually required http tokens to access the ami.

File: variables.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (betabandido) [reply variables.tf:62]: @xorxsan Added some description.

**Pull request merged**